### ISSUE TITLE
Failure to validate templates that use the AWS::URLSuffix psedo parameter

### ISSUE THREAD

**Issue by nskoufis** [martysweet/cfn-lint#173]
When validating templates that include the `AWS::URLSuffix` pseudo parameter, the value is not recognised and an error is thrown.

Steps to reproduce:
1. Execute `cfn-lint validate sample.json`, where `sample.json` contains the following:
```json
{
    "Resources": {
        "TestLambdaPermissionEventsRuleSchedule1": {
            "Type": "AWS::Lambda::Permission",
            "Properties": {
                "FunctionName": "test",
                "Action": "lambda:InvokeFunction",
                "Principal": {
                    "Fn::Join": [
                        "",
                        [
                            "events.",
                            {
                                "Ref": "AWS::URLSuffix"
                            }
                        ]
                    ]
                },
                "SourceArn": "arn:aws:lambda:us-east-1:123456789012:function:Test"
            }
        }
    }
}
```

Expected outcome: no validation errors
Actual outcome:
```
0 infos
0 warn
1 crit
Resource: Resources > TestLambdaPermissionEventsRuleSchedule1 > Properties > Principal
Message: Referenced value AWS::URLSuffix does not exist
Documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html

Template invalid!
```

I'm working on a PR for this. It's a simple fix, I just need some tests

**Comment by martysweet** [comment]
This should be fixed in v.1.7.3 which is now on npm.

**Comment by martysweet** [comment]
Thanks for your help!
